Userland/Libraries/LibJS/Heap/Heap.cpp

```diff
@@ -274,6 +274,48 @@ void Heap::sweep_dead_cells(bool print_report, const Core::ElapsedTimer& measure
     }
 }
 
+void Heap::force_collect(Object const& object)
```

Comment:
I don't like having this dangerous method in Heap but could not think of any way of protecting it...

Reply:
Would something like `Badge<TestGlobalObject>` work here? (I'm not sure if `TestGlobalObject` is actually the class that calls `force_collect`, so not sure if this works).

Reply:
I don't think it's actually called from TestRunnerGlobalObject unfortunately and also it feels a bit strange to have a dependency on the test program.

Reply:
Can we not duplicate all of the garbage collection code just for a single test-only method? We can add a "inverse-root" (or a better name) that we can take into account when collecting cells for normal garbage collection, and then just trigger said normal garbage collection.

Reply:
That would be cleaner, let me try that.

Reply:
BTW, while you're here, we disabled the FinalizationRegistry test for the same issue i think, so after you fix this you can probably re-enable it as well